backend/main.go

```diff
@@ -138,78 +200,144 @@ func roll(level int, rolls int) {
 			}
 
 			if c, ok := rollResults[champion]; ok {
-				rollResults[champion] = rollResult{champion, cost + 1, c.Appearances + 1}
+				rollResults[champion] = rollResult{champion, cost + 1, c.Appearances + 1, ""}
 			} else {
-				rollResults[champion] = rollResult{champion, cost + 1, 1}
+				rollResults[champion] = rollResult{champion, cost + 1, 1, ""}
 			}
 		}
 	}
+
+	// calculate the percentage of appearance for each champion
+	for i, s := range rollResults {
+		p := (float32(s.Appearances) * 100) / (float32(rolls) * 5)
+		s.Percentage = fmt.Sprintf("%.2f", p) + "%"
+		rollResults[i] = s
+	}
 }
 
 func rollLevel(w http.ResponseWriter, r *http.Request) {
 	// sanitize input variables
 	levelStr := r.URL.Query().Get("level") // returns empty string if not found
 	level, err := strconv.Atoi(strings.Trim(levelStr, " "))
 	if err != nil {
-		w.WriteHeader(400)
-		w.Write([]byte("Incomplete request: provide a valid level."))
-		log.Fatalf("Error: invalid level.")
+		w.WriteHeader(http.StatusBadRequest)
+		w.Header().Set("Content-Type", "application/json")
+		resp := make(map[string]string)
+		resp["message"] = "Bad Request: invalid level"
+		jsonResp, err := json.Marshal(resp)
+		if err != nil {
+			fmt.Printf("Error happened in JSON marshal. Err: %s", err)
+		}
+		w.Write(jsonResp)
 		return
 	}
 
 	rollNumsStr := r.URL.Query().Get("rolls") // number of rolls to perform
 	rollNums, err := strconv.Atoi(strings.Trim(rollNumsStr, " "))
 
 	if err != nil || rollNums > 200 {
-		w.WriteHeader(400)
-		w.Write([]byte("Incomplete request: provide a valid number of rolls (number less than 200)."))
-		log.Fatalf("Error: invalid number of rolls.")
+		w.WriteHeader(http.StatusBadRequest)
+		w.Header().Set("Content-Type", "application/json")
+		resp := make(map[string]string)
```

Comment:
This can be simplified using a map literal

````go
resp := map[string]string{
  "message": "Bad Request: Provide a valid number of rolls (number less than 200)"
}
```